lib/puppet/pops/model/factory.rb

```diff
@@ -89,6 +90,29 @@ def initialize(o, *args)
     end
   end
 
+  def map_offset(model, locator)
+    MAPOFFSET_VISITOR.visit_this_1(self, model, locator)
+  end
+
+  def map_offset_Object(o, locator)
```

Comment:
I couldn't find any other reference to this method; is this supposed to be some sort of interface that is implemented elsewhere?

Reply:
@tvpartytonight It is called from a polymorphic dispatcher - it calls methods having the pattern `map_<classname>`. In order to not error when there is no more specific rule, there must be a `map_Object` which is the catch all / "else" branch of the logic. Read about Polymorphic dispatch here: http://puppet-on-the-edge.blogspot.com/2014/02/puppet-internals-polymorphic-dispatch.html

Reply:
See line 31 where the `map_offset` dispatcher (a.k.a Visitor) is configured, and line 94 where it is used - i.e. calling one of the `map_offset_<class>` methods.